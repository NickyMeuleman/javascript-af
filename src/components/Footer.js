import React from 'react'
import Footer from './styles/Footer'

export default class Foot extends React.Component {
  render () {
    const links = [
      {
        name: 'About',
        href: '/about'
      },
      {
        name: 'Team',
        href: '/team'
      },
      {
        name: 'Jobs',
        href: '/jobs'
      },
      {
        name: 'Terms',
        href: '/terms'
      },
      {
        name: 'Privacy Policy',
        href: '/privacy'
      }
    ]
    const creators = [
      {
        name: 'Jesse',
        href: 'https://www.instagram.com/jesse.weigel/'
      },
      {
        name: 'Jvscrpt.r',
        href: 'https://www.instagram.com/jvscrptr/'
      },
      // There is a reason for me doing this just trust me you'll see later on
      /*{
        name: 'Harshit',
        href: 'https://www.instagram.com/pantharshit00/'
      }*/
    ]
    return (
      <Footer>
        <p>
          Made with <span id="heart">&#9829;</span> by{' '}
          {creators.map((item, i) => (
            <React.Fragment key={`creator-${item.name}`}>
              <a href={item.href} id="name">
                {item.name}
              </a>
              {i === creators.length - 2
                ? ' and '
                : i === creators.length - 1 ? '' : ' ,'}
            </React.Fragment>
          ))}
        </p>
        {links.map(item => (
          <a href={item.href} key={`footer-link-${item.name}`}>
            {item.name}
          </a>
        ))}
      </Footer>
    )
  }
}
