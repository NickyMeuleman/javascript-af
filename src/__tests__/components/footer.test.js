/* eslint-env jest */
import React from 'react'
import render from 'react-test-renderer'
import Footer from '../../components/Footer'

describe('<Footer />', () => {
  it('should render correctly', () => {
    const tree = render.create(<Footer />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
