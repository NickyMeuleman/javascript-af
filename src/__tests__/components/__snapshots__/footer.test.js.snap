// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<div
  className="sc-bdVaJa cZKaVR"
>
  <p>
    Made with 
    <span
      id="heart"
    >
      â™¥
    </span>
     by
     
    <a
      href="https://www.instagram.com/jesse.weigel/"
      id="name"
    >
      Jesse
    </a>
     ,
    <a
      href="https://www.instagram.com/jvscrptr/"
      id="name"
    >
      Jvscrpt.r
    </a>
     and 
    <a
      href="https://www.instagram.com/pantharshit00/"
      id="name"
    >
      Harshit
    </a>
    
  </p>
  <a
    href="/about"
  >
    About
  </a>
  <a
    href="/jobs"
  >
    Jobs
  </a>
  <a
    href="/terms"
  >
    Terms
  </a>
  <a
    href="/privacy"
  >
    Privacy Policy
  </a>
</div>
`;
